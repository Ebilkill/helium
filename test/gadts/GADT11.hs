data X a where
    A :: (b -> Int) -> b -> X Int

f :: X a -> a
f (A f x) = x
