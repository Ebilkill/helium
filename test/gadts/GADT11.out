Prelude is up to date
Compiling gadts/GADT11.hs
(5,6): Warning: Variable "f" is not used
(5,6): Warning: Variable "f" shadows the one at (5,1)
fromList []
Unifying non-touchable variables [_10,_14]


(4,1): Type signature is too general
 function         : f
   declared type  : X a   -> a  
   inferred type  : X Int -> Int

Compilation failed with 1 error
